stackslib/src/chainstate/stacks/mod.rs

```diff
@@ -621,13 +619,57 @@ impl_array_hexstring_fmt!(TokenTransferMemo);
 impl_byte_array_newtype!(TokenTransferMemo, u8, 34);
 impl_byte_array_serde!(TokenTransferMemo);
 
+/// Cause of change in mining tenure
+/// Depending on cause, tenure can be ended or extended
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum TenureChangeCause {
+    /// A valid winning block-commit, end current tenure
+    BlockFound = 0,
+    /// No winning block-commits, extend current tenure
+    NoBlockFound = 1,
+    /// A “null miner” won the block-commit (see the MEV solution below)
+    NullMiner = 2,
+}
+
+impl TryFrom<u8> for TenureChangeCause {
+    type Error = ();
+
+    fn try_from(num: u8) -> Result<Self, Self::Error> {
+        match num {
+            0 => Ok(Self::BlockFound),
+            1 => Ok(Self::NoBlockFound),
+            2 => Ok(Self::NullMiner),
+            _ => Err(()),
+        }
+    }
+}
+
+/// A transaction from Stackers to signal new mining tenure
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct TenureChangePayload {
+    /// The StacksBlockId of the last block from the previous tenure
+    pub previous_tenure_end: StacksBlockId,
+    /// The number of blocks produced in the previous tenure
+    pub previous_tenure_blocks: u16,
+    /// A flag to indicate which of the following triggered the tenure change
+    pub cause: TenureChangeCause,
+    /// The ECDSA public key hash of the current tenure
+    pub pubkey_hash: Hash160,
+    /// A Schnorr signature from at least 70% of the Stackers
+    //pub signature: ???
```

Comment:
CC @xoloki

Reply:
A taproot schnorr signature is 64 bytes, using the x-only curve point format.  A regular schnorr proof using secp256k1 would be 65 bytes, with a 33 byte compressed curve point, or 96 bytes with an uncompressed curve point.

Reply:
This is the standard `R,s` schnorr proof where `R` is a curve point and `s` is a scalar.

Reply:
No reason to allow different formats, I'm just letting y'all know what the options are.  If we're using `wsts` to make a distributed threshold signature, then we get either the 65 or 64 byte formats.  If we're using a libsecp256k1 wrapper for a non distributed threshold signature, then we'll only get the 64 byte format.

Reply:
Which we should use depends on what tradeoffs we care about.  If we're going to use `wsts` to make a distributed signature, and also verify it, then the 65 byte format is the cleanest, since it doesn't require jumping through all the hoops necessary for the 64 byte format.  If we care about saving the extra byte, or plan on using a `libsecp256k1` wrapper to verify, then we need to use the 64 byte format.

Reply:
I'm in favor of the 65 byte format in `wsts`.

Reply:
Agreed, I think that's the best choice for block signing for `Nakamoto`.  `sBTC` will want to use the 64 byte taproot format, since they need to sign `BTC` tx.

Reply:
So a signature in `TenureChange` should consist of both a point and scalar from `wsts`, like this?

```rust
struct SchnorrThresholdSignature {
    pub point: wsts::Point,
    pub scalar: wsts::Scalar,
}
```

Reply:
Yes, that works perfectly.  If you want a byte representation, you can call

`point.compress().as_bytes()`
`scalar.to_bytes()`

Reply:
There's also a `wsts::common::Signature` struct that you can use instead of defining your own.

Reply:
Thanks, that's exactly what I need